service/topology_guard.hh

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C) 2023-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+
+#include "service/session.hh"
+#include "replica/database_fwd.hh"
+
+namespace service {
+
+/// Represents topology_guard in transit.
+/// Can be passed across nodes and copied across shards.
+/// Guard in transit doesn't block barriers, but can be materialized into topology_guard
+/// which does. If the guard was invalidated while in transit, materialization will fail.
+/// Has an null state which represents no guard.
+/// topology_guard created out of null frozen_topology_guard is valid and never aborted.
+using frozen_topology_guard = session_id;
```

Comment:
Nitpick but do we really need the type? Why not use session_id directly? Or, if you think we will need to extend it at some point, we can make session_id to be a struct with its own extensible IDL. IMO as it is `frozen_topology_guard ` provides nothing but obfuscation. It does not make the type extensible on wire.

Reply:
Why not simple make `topology_guard` serializable?

Reply:
From my experience with rpc, it is better to use a class for on wire objects (especially for the one that rpc verb returns) so we can extend later just in case. 

class frozen_topology_guard {
       session_id id;
}

Reply:
> Why not simple make `topology_guard` serializable?

frozen_topology_guard is not just to be able to serialize the guard, it is another form of the guard which has slightly different semantics. You can copy it across shards and it doesn't hold the coordinator (yet).